### PR TITLE
python3 for CI

### PR DESCRIPTION
**Pull request by bam241**
Updating docker file to use python3.

it is probably time we retire python2

### CODE REVIEW

**Comment by bam241** [comment]
shall we upload temporary docker image to test this ?

File: CI/Dockerfile_0_base [hunk 1]
```diff
@@ -6,10 +6,29 @@ SHELL ["/bin/bash", "-c"]
...
-                       python-numpy python-pip python-setuptools wget; \
-    pip install cython
+    apt-get -y install autoconf \
+                       clang \
+                       cmake \
+                       g++ \
+                       gcc \
+                       gfortran \
+                       git \
+                       libblas-dev \
```
Comment (gonuke) [inline CI/Dockerfile_0_base:16]: Unrelated to this PR, but I don't think we need BLAS/LAPACK anymore...?

**Review by gonuke — COMMENTED** [review]
LGTM - if we can get things tested, I'll merge

**Comment by bam241** [comment]
@gonuke it is building against the newly pushed image (repushed 15 minutes ago)

this should be redo when CI passes

File: CI/Dockerfile_0_base (hunk 1 above)
Comment (bam241) [reply CI/Dockerfile_0_base:16]: I forgot about this, issuing a news PR for it !
(maybe it could be a good test for our docker building/testing PR)

**Comment by gonuke** [comment]
Thanks @bam241 - I've made an issue (#750) to review the image dependencies.

**Pull request merged**